sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -160,6 +160,63 @@ static rom_error_t boot_data_entry_read(uint32_t page_base, size_t index,
   return kErrorOk;
 }
 
+/**
+ * Populates the boot data entry at the given page and index.
+ *
+ * @param page_base Page base address in bytes.
+ * @param index Index of the entry to write in the given page.
+ * @param boot_data Entry to write.
+ * @return The result of the operation.
+ */
+static rom_error_t boot_data_entry_write(uint32_t page_base, size_t index,
+                                         const boot_data_t *boot_data) {
+  const uint32_t addr = boot_data_entry_address_get(page_base, index);
+  boot_data_buffer_t buf;
+  memcpy(&buf, boot_data, sizeof(boot_data_t));
+  // TODO(#8777): Update error handling after switching to silicon_creator
+  // driver.
+  if (flash_write(addr, kInfoPartition, buf.data, kBootDataNumWords) != 0) {
```

Comment:
IIUC this would help with interrupted writes, right? We already have a digest and the previous entry is invalidated only after the write succeeds. We can definitely make this change but I'm not sure what the benefit would be in this case. WDYT?